packages/openapi3/test/test-return-types.ts

```diff
@@ -150,6 +307,95 @@ describe("openapi3: return types", () => {
     strictEqual(diagnostics[0].message, "Duplicate @body declarations on response type");
   });
 
+  it("issues diagnostics for return type with duplicate status code", async () => {
+    const diagnostics = await checkFor(
+      `
+    model Foo {
+      foo: string;
+    }
+    model Error {
+      code: string;
+    }
+    @route("/")
+    namespace root {
+      @get
+      op read(): Foo | Error;
+    }
+    `
+    );
+    strictEqual(diagnostics.length, 1);
+    strictEqual(diagnostics[0].code, "@cadl-lang/openapi3/duplicate-response");
+    strictEqual(diagnostics[0].message, "Multiple return types for status code 200");
+  });
+
+  it("issues diagnostics for invalid status codes", async () => {
+    const diagnostics = await checkFor(
+      `
+      model Foo {
+        foo: string;
+      }
+
+      model BadRequest {
+        code: "4XX";
+      }
+
+      namespace root {
+        @route("/test1")
+        @get
+        op test1(): { @statusCode code: string, @body body: Foo };
+        @route("/test2")
+        @get
+        op test2(): { @statusCode code: "Ok", @body body: Foo };
+        @route("/test3")
+        @get
+        op test3(): { @statusCode code: "200" | BadRequest, @body body: Foo };
+      }
+    `
+    );
+    strictEqual(diagnostics.length, 3);
+    ok(diagnostics.every((e) => e.code === "@cadl-lang/openapi3/status-code-invalid"));
+    ok(diagnostics[0].message.includes("must be a numeric or string literal"));
```

Comment:
Feel like it might be nice to get an assert library. The built-in `assert` is quite limited and we result in having a lot of `ok` which give bad error message if it fails.

Reply:
I'd be for that.  Another option could be to just require an error message on `ok`.